### PR TITLE
Version/3.3

### PR DESCRIPTION
**Pull request by mejango**
- fixed voting units lock bug.
- renamed totalSupply to totalSupplyOf to be consistent with the naming style of other functions.

### CODE REVIEW

File: contracts/JBTiered721Delegate.sol [hunk 1]
```diff
@@ -1,776 +1,632 @@
...
-
-  @dev
-  Inherits from -
-  JB721Delegate: A generic NFT delegate.
-  Votes: A helper for voting balance snapshots.
-  JBOwnable: Includes convenience functionality for checking a message sender's permissions before executing certain transactions.
-*/
+import {IJBOperatorStore} from "@jbx-protocol/juice-contracts-v3/contracts/abstract/JBOperatable.sol";
+import {JBOwnable, JBOwnableOverrides} from "@jbx-protocol/juice-ownable/src/JBOwnable.sol";
+import {JB721Operations} from "./libraries/JB721Operations.sol";
```
Comment (drgorillamd) [inline contracts/JBTiered721Delegate.sol:6]: nit> sort ext/internal import + follow the import { XX } from "YY"

File: contracts/JBTiered721Delegate.sol [hunk 2]
```diff
@@ -1,776 +1,632 @@
...
+        IJBFundingCycleStore _fundingCycleStore,
+        string memory _baseUri,
+        IJB721TokenUriResolver _tokenUriResolver,
+        string memory _contractUri,
+        JB721PricingParams memory _pricing,
+        IJBTiered721DelegateStore _store,
+        JBTiered721Flags memory _flags
+    ) public override {
+        // Make the original un-initializable.
+        if (address(this) == codeOrigin) revert();
```
Comment (drgorillamd) [inline contracts/JBTiered721Delegate.sol:183]: Is this needed? There is no selfdestruct, having an initialised original doesn't change much

File: contracts/JBTiered721Delegate.sol [hunk 3]
```diff
@@ -1,776 +1,632 @@
...
-    // Record the minted reserves for the tier.
-    uint256[] memory _tokenIds = store.recordMintReservesFor(_tierId, _count);
+        // Record adding the provided tiers.
+        if (_pricing.tiers.length != 0) _store.recordAddTiers(_pricing.tiers);
 
-    // Keep a reference to the reserved token beneficiary.
-    address _reservedTokenBeneficiary = store.reservedTokenBeneficiaryOf(address(this), _tierId);
+        // Set the flags if needed.
+        if (
+            _flags.lockReservedTokenChanges || _flags.lockVotingUnitChanges || _flags.lockManualMintingChanges
```
Comment (drgorillamd) [inline contracts/JBTiered721Delegate.sol:220]: If we want to golf a bit, we could just mload the struct (as it's in mem) then check if not zero, but maybe overshooting

File: contracts/JBTiered721Delegate.sol (hunk 2 above)
Comment (mejango) [reply contracts/JBTiered721Delegate.sol:183]: deleted

**Review by viraj124 — APPROVED** [review]

**Pull request merged**